verilog/parser/verilog.lex

```diff
@@ -672,9 +672,8 @@ wait_order { UpdateLocation(); return TK_wait_order; }
 wildcard { UpdateLocation(); return TK_wildcard; }
 <COVERGROUP>with { UpdateLocation(); return TK_with__covergroup; }
 with { UpdateLocation(); return TK_with; }
-"with"[ \t]*"[" {
-  /* TODO: formulate the ws sequence as any whitespace+comment sequence*/
-  yyless(4);
+"with"({TraditionalCommentOrSpace}|{EndOfLineComment})*"[" {
```

Comment:
This might be easier with an `<AFTER_WITH>` state, where `with` returns a `TK_width` after pushing a new state.
It will be easier to re-use rules inside states than glue them together like this in monolithic regular expressions.
Inside this new state, you handle whitespace, comments, and `[`, and for everything else, you can pop the state and rewind, to defer to default `<INITIAL>` state handling (or whatever was previously on the state stack).

Reply:
You will also want lexer unit-tests for this.  :)

Reply:
Ok, will need to familiarize myself how states work. They did look somewhat complicated.

The beauty if this is, that I don't have to handle anything special, including '['; I just look ahead, find that this is the next token and TK_with needs to be a new token.

Reply:
(lexer unit test added)

Reply:
I looked at the AFTER_DOT example, which is somewhat involved. I have to play with it to see if it is more advantageous.

Reply:
Looking over this again, I still stand by this first comment.  Generally, the options are:

1) track state within the lexer to alter the enumeration of the _latter_ token
2) track state inside the lexical context class to alter the enumeration of the _earlier_ token (effectively achieving lookahead) -- this approach is pretty localized, you're defining a tiny state-machine that can run along side other independent state machines.